### ISSUE TITLE
Improve logging of JavaScript files type-checking in SonarLint context

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#4186]
Depends on #4190

The type-checking of JavaScript files in SonarLint context is deactivated when the project size exceeds a given threshold. Currently, users are warned about this deactivation with the following log:

```
Project type checking for JavaScript files deactivated as project has too many files (maximum is 20000 files)
````

The log should emphasize the consequence of type-checking deactivation: JavaScript rules using type information might not behave as expected.

Furthermore, since the log mentions the value of the threshold, it should also mention the actual project size so that users know to what value they could increase the threshold if necessary.

**Comment by ilia-kebets-sonarsource** [comment]
related to https://github.com/SonarSource/SonarJS/issues/4187
